### ISSUE TITLE
1.2 kontena stack upgrade deploys services multiple times

### ISSUE THREAD

**Issue by SpComb** [kontena/kontena#2081]
```
I, [2017-04-05T16:44:58.793470 #7]  INFO -- GridServiceDeployer: starting to deploy test/wordpress/mysql
I, [2017-04-05T16:44:58.963506 #7]  INFO -- GridServiceDeployer: starting to deploy test/wordpress/wordpress
I, [2017-04-05T16:44:58.994312 #7]  INFO -- GridServiceDeployer: throttling service instance test/wordpress/mysql deploy because of min_health limit (1 instances in-progress)
I, [2017-04-05T16:44:59.019888 #7]  INFO -- GridServiceInstanceDeployer: Deploying service instance test/wordpress/mysql-1 to node weathered-resonance-64 at 2017-04-05 16:44:58 UTC...
82.181.164.219 - - [05/Apr/2017:16:44:59 +0000] "PUT /v1/stacks/test/wordpress HTTP/1.1" 200 4185 0.4346
I, [2017-04-05T16:44:59.236006 #7]  INFO -- GridServiceDeployer: throttling service instance test/wordpress/wordpress deploy because of min_health limit (1 instances in-progress)
I, [2017-04-05T16:44:59.236767 #7]  INFO -- GridServiceInstanceDeployer: Deploying service instance test/wordpress/wordpress-1 to node weathered-resonance-64 at 2017-04-05 16:44:58 UTC...
I, [2017-04-05T16:45:00.029778 #7]  INFO -- StackDeployWorker: removing following services: 
82.181.164.219 - - [05/Apr/2017:16:45:00 +0000] "POST /v1/stacks/test/wordpress/deploy HTTP/1.1" 200 140 0.2918

I, [2017-04-05T16:45:03.496988 #7]  INFO -- GridServiceDeployer: service test/wordpress/wordpress has been deployed

I, [2017-04-05T16:45:05.933404 #7]  INFO -- GridServiceDeployer: service test/wordpress/mysql has been deployed

I, [2017-04-05T16:45:05.998505 #7]  INFO -- GridServiceSchedulerWorker: delaying test/wordpress/mysql deploy because there is another deploy in progress
I, [2017-04-05T16:45:06.077033 #7]  INFO -- GridServiceSchedulerWorker: delaying test/wordpress/mysql deploy because there is another deploy in progress

I, [2017-04-05T16:45:36.136043 #7]  INFO -- GridServiceDeployer: starting to deploy test/wordpress/mysql
I, [2017-04-05T16:45:36.663813 #7]  INFO -- GridServiceDeployer: throttling service instance test/wordpress/mysql deploy because of min_health limit (1 instances in-progress)
I, [2017-04-05T16:45:36.668577 #7]  INFO -- GridServiceInstanceDeployer: Deploying service instance test/wordpress/mysql-1 to node weathered-resonance-64 at 2017-04-05 16:45:36 UTC...

I, [2017-04-05T16:45:38.583394 #7]  INFO -- GridServiceDeployer: service test/wordpress/mysql has been deployed

I, [2017-04-05T16:45:38.662452 #7]  INFO -- GridServiceDeployer: starting to deploy test/wordpress/mysql
I, [2017-04-05T16:45:38.792288 #7]  INFO -- GridServiceDeployer: throttling service instance test/wordpress/mysql deploy because of min_health limit (1 instances in-progress)
I, [2017-04-05T16:45:38.795048 #7]  INFO -- GridServiceInstanceDeployer: Deploying service instance test/wordpress/mysql-1 to node weathered-resonance-64 at 2017-04-05 16:45:38 UTC...
82.181.164.219 - - [05/Apr/2017:16:45:40 +0000] "GET /v1/stacks/test/wordpress/deploys/58e51f0b938003000700052a HTTP/1.1" 200 721 0.2814
I, [2017-04-05T16:45:40.607018 #7]  INFO -- GridServiceDeployer: starting to deploy test/wordpress/wordpress
I, [2017-04-05T16:45:40.706865 #7]  INFO -- GridServiceDeployer: throttling service instance test/wordpress/wordpress deploy because of min_health limit (1 instances in-progress)
I, [2017-04-05T16:45:40.709645 #7]  INFO -- GridServiceInstanceDeployer: Deploying service instance test/wordpress/wordpress-1 to node weathered-resonance-64 at 2017-04-05 16:45:40 UTC...
I, [2017-04-05T16:45:40.739900 #7]  INFO -- GridServiceDeployer: service test/wordpress/mysql has been deployed

I, [2017-04-05T16:45:42.526473 #7]  INFO -- GridServiceDeployer: service test/wordpress/wordpress has been deployed

```

Mongo dumps:

`db.stack_deploys.find()`
```
{ "_id" : ObjectId("58e51f0b938003000700052a"), "_deploy_state" : "success", "stack_id" : ObjectId("58e51284e8140a000600006a"), "updated_at" : ISODate("2017-04-05T16:45:42.827Z"), "created_at" : ISODate("2017-04-05T16:44:59.994Z") }
```

`db.grid_service_deploys.find({"stack_deploy_id" : ObjectId("58e51f0b938003000700052a")})`
```
{ "_id" : ObjectId("58e51f0c938003000700052c"), "_deploy_state" : "success", "grid_service_id" : ObjectId("58e51284e8140a000600006c"), "updated_at" : ISODate("2017-04-05T16:45:00.055Z"), "created_at" : ISODate("2017-04-05T16:45:00.055Z"), "stack_deploy_id" : ObjectId("58e51f0b938003000700052a"), "started_at" : ISODate("2017-04-05T16:45:36.067Z"), "grid_service_instance_deploys" : [ { "_id" : ObjectId("58e51f309380030007000536"), "_deploy_state" : "success", "instance_number" : 1, "host_node_id" : ObjectId("58d11c81a0ff6300060001e5"), "updated_at" : ISODate("2017-04-05T16:45:36.644Z"), "created_at" : ISODate("2017-04-05T16:45:36.644Z") } ], "finished_at" : ISODate("2017-04-05T16:45:38.583Z") }
{ "_id" : ObjectId("58e51f339380030007000539"), "_deploy_state" : "success", "grid_service_id" : ObjectId("58e51284e8140a0006000071"), "updated_at" : ISODate("2017-04-05T16:45:39.607Z"), "created_at" : ISODate("2017-04-05T16:45:39.607Z"), "stack_deploy_id" : ObjectId("58e51f0b938003000700052a"), "started_at" : ISODate("2017-04-05T16:45:40.590Z"), "grid_service_instance_deploys" : [ { "_id" : ObjectId("58e51f34938003000700053f"), "_deploy_state" : "success", "instance_number" : 1, "host_node_id" : ObjectId("58d11c81a0ff6300060001e5"), "updated_at" : ISODate("2017-04-05T16:45:40.702Z"), "created_at" : ISODate("2017-04-05T16:45:40.702Z") } ], "finished_at" : ISODate("2017-04-05T16:45:42.526Z") }
```

`db.grid_service_deploys.find({"grid_service_id": ObjectId("58e51284e8140a000600006c")})` (mysql)
```
{ "_id" : ObjectId("58e51f099380030007000521"), "_deploy_state" : "success", "grid_service_id" : ObjectId("58e51284e8140a000600006c"), "updated_at" : ISODate("2017-04-05T16:44:57.976Z"), "created_at" : ISODate("2017-04-05T16:44:57.976Z"), "started_at" : ISODate("2017-04-05T16:44:58.760Z"), "grid_service_instance_deploys" : [ { "_id" : ObjectId("58e51f0a9380030007000527"), "_deploy_state" : "success", "instance_number" : 1, "host_node_id" : ObjectId("58d11c81a0ff6300060001e5"), "updated_at" : ISODate("2017-04-05T16:44:58.973Z"), "created_at" : ISODate("2017-04-05T16:44:58.973Z") } ], "finished_at" : ISODate("2017-04-05T16:45:05.938Z") }
{ "_id" : ObjectId("58e51f0c938003000700052c"), "_deploy_state" : "success", "grid_service_id" : ObjectId("58e51284e8140a000600006c"), "updated_at" : ISODate("2017-04-05T16:45:00.055Z"), "created_at" : ISODate("2017-04-05T16:45:00.055Z"), "stack_deploy_id" : ObjectId("58e51f0b938003000700052a"), "started_at" : ISODate("2017-04-05T16:45:36.067Z"), "grid_service_instance_deploys" : [ { "_id" : ObjectId("58e51f309380030007000536"), "_deploy_state" : "success", "instance_number" : 1, "host_node_id" : ObjectId("58d11c81a0ff6300060001e5"), "updated_at" : ISODate("2017-04-05T16:45:36.644Z"), "created_at" : ISODate("2017-04-05T16:45:36.644Z") } ], "finished_at" : ISODate("2017-04-05T16:45:38.583Z") }
{ "_id" : ObjectId("58e51f0a9380030007000523"), "_deploy_state" : "success", "grid_service_id" : ObjectId("58e51284e8140a000600006c"), "updated_at" : ISODate("2017-04-05T16:44:58.355Z"), "created_at" : ISODate("2017-04-05T16:44:58.355Z"), "started_at" : ISODate("2017-04-05T16:45:36.524Z"), "grid_service_instance_deploys" : [ { "_id" : ObjectId("58e51f329380030007000538"), "_deploy_state" : "success", "instance_number" : 1, "host_node_id" : ObjectId("58d11c81a0ff6300060001e5"), "updated_at" : ISODate("2017-04-05T16:45:38.786Z"), "created_at" : ISODate("2017-04-05T16:45:38.786Z") } ], "finished_at" : ISODate("2017-04-05T16:45:40.740Z") }
```

`db.grid_service_deploys.find({"grid_service_id": ObjectId("58e51284e8140a0006000071")})` (wordpress)
```
{ "_id" : ObjectId("58e51f0a9380030007000524"), "_deploy_state" : "success", "grid_service_id" : ObjectId("58e51284e8140a0006000071"), "updated_at" : ISODate("2017-04-05T16:44:58.359Z"), "created_at" : ISODate("2017-04-05T16:44:58.359Z"), "started_at" : ISODate("2017-04-05T16:44:58.923Z"), "grid_service_instance_deploys" : [ { "_id" : ObjectId("58e51f0b9380030007000528"), "_deploy_state" : "success", "instance_number" : 1, "host_node_id" : ObjectId("58d11c81a0ff6300060001e5"), "updated_at" : ISODate("2017-04-05T16:44:59.218Z"), "created_at" : ISODate("2017-04-05T16:44:59.218Z") } ], "finished_at" : ISODate("2017-04-05T16:45:03.497Z") }
{ "_id" : ObjectId("58e51f339380030007000539"), "_deploy_state" : "success", "grid_service_id" : ObjectId("58e51284e8140a0006000071"), "updated_at" : ISODate("2017-04-05T16:45:39.607Z"), "created_at" : ISODate("2017-04-05T16:45:39.607Z"), "stack_deploy_id" : ObjectId("58e51f0b938003000700052a"), "started_at" : ISODate("2017-04-05T16:45:40.590Z"), "grid_service_instance_deploys" : [ { "_id" : ObjectId("58e51f34938003000700053f"), "_deploy_state" : "success", "instance_number" : 1, "host_node_id" : ObjectId("58d11c81a0ff6300060001e5"), "updated_at" : ISODate("2017-04-05T16:45:40.702Z"), "created_at" : ISODate("2017-04-05T16:45:40.702Z") } ], "finished_at" : ISODate("2017-04-05T16:45:42.526Z") }
```

**Comment by SpComb** [comment]
This is because the stack secrets `PUT` from the stack variables resolver forces a service update:

```
I, [2017-04-06T15:01:24.530486 #624]  INFO -- GridSecrets::Update: force service development/wordpress/mysql update @
	/home/kontena/kontena/kontena/server/vendor/bundle/ruby/2.3.0/gems/mongoid-4.0.2/lib/mongoid/contextual/mongo.rb:658:in `yield_document'
	/home/kontena/kontena/kontena/server/vendor/bundle/ruby/2.3.0/gems/mongoid-4.0.2/lib/mongoid/contextual/mongo.rb:123:in `block in each'
	/home/kontena/kontena/kontena/server/vendor/bundle/ruby/2.3.0/gems/moped-2.0.7/lib/moped/cursor.rb:29:in `block in each'
	/home/kontena/kontena/kontena/server/vendor/bundle/ruby/2.3.0/gems/moped-2.0.7/lib/moped/cursor.rb:29:in `each'
	/home/kontena/kontena/kontena/server/vendor/bundle/ruby/2.3.0/gems/moped-2.0.7/lib/moped/cursor.rb:29:in `each'
	/home/kontena/kontena/kontena/server/vendor/bundle/ruby/2.3.0/gems/moped-2.0.7/lib/moped/query.rb:78:in `each'
	/home/kontena/kontena/kontena/server/vendor/bundle/ruby/2.3.0/gems/mongoid-4.0.2/lib/mongoid/contextual/mongo.rb:122:in `each'
	/home/kontena/kontena/kontena/server/vendor/bundle/ruby/2.3.0/gems/mongoid-4.0.2/lib/mongoid/contextual.rb:20:in `each'
	/home/kontena/kontena/kontena/server/app/mutations/grid_secrets/common.rb:9:in `refresh_grid_services'
	/home/kontena/kontena/kontena/server/app/mutations/grid_secrets/update.rb:21:in `execute'
	/home/kontena/kontena/kontena/server/vendor/bundle/ruby/2.3.0/gems/mutations-0.7.2/lib/mutations/command.rb:80:in `run'
	/home/kontena/kontena/kontena/server/vendor/bundle/ruby/2.3.0/gems/mutations-0.7.2/lib/mutations/command.rb:32:in `run'
	/home/kontena/kontena/kontena/server/app/routes/v1/secrets.rb:47:in `update_secret'
	/home/kontena/kontena/kontena/server/app/routes/v1/secrets.rb:75:in `block (4 levels) in <class:SecretsApi>'
```

**Comment by SpComb** [comment]
Fixed by #2095
